instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java

```diff
@@ -116,6 +125,22 @@ public void mapLogEvent(
       attributes.put(THREAD_ID, threadId);
     }
 
+    if (captureCodeAttributes) {
+      StackTraceElement source = sourceSupplier.get();
+      if (source != null) {
+        String fileName = source.getFileName();
+        if (fileName != null) {
```

Comment:
(optional)

```suggestion
        // file name can be null from the StackTraceElement Javadoc
        if (fileName != null) {
```

Reply:
This is copy-pasted from the logback instrumentation. Apparently you are the author of it.